nntrainer/graph/graph_core.h

```diff
@@ -114,16 +117,19 @@ class GraphCore {
 
   /**
    * @brief     get end iterator for the forwarding
-   * @retval    const iterator marking the emd of forwarding
+   * @retval    const iterator marking the end of forwarding
+   * @note      this function should not be used when node_list is empty.
+   * if node_list is empty, end iterator is dereferenced,
```

Comment:
how about we check the node_list.empty() and do something before return.

Reply:
Usually, empty check is user's responsiblity.
https://cplusplus.com/reference/vector/vector/cbegin/
```suggestion
   * if node_list is empty, end iterator shall not be dereferenced,
```